src/fmelt.c

```diff
@@ -302,6 +303,11 @@ static void preprocess(SEXP DT, SEXP id, SEXP measure, SEXP varnames, SEXP valna
   SEXPTYPE type;
   data->lmax = 0; data->totlen = 0; data->nrow = length(VECTOR_ELT(DT, 0));
   SET_VECTOR_ELT(data->RCHK, 0, vars = checkVars(DT, id, measure, verbose));
+  if(!isNull(measure) && isNewList(measure)){
+    data->measure_is_list = TRUE;
+  }else{
+    data->measure_is_list = FALSE;
+  }
```

Comment:
```suggestion
  data->measure_is_list = !isNull(measure) && isNewList(measure) ? TRUE : FALSE;
```

Alternatively, this could be a simple `bool` in the struct and we just set `data->measure_is_list = !isNull(measure) && isNewList(measure)`.